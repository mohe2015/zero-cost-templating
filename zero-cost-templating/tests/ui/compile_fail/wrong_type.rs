#![deny(warnings)]
#![feature(coroutines)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub async fn test() {
    let template = test_initial0();
    let template = template.test_template0();
    let page_title = Cow::from("thetitle");
    let template = template.test_page_title1(page_title);
    let csrf_token = Cow::from("thetoken");
    let template = template.test_csrf_token2(csrf_token);
    let template = template.test_template6();
    let test: String = { template.test_copyright_year7(Cow::from("2023".to_string())) };
}

pub fn main() {}
