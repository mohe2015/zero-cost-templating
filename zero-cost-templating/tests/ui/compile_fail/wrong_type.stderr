error[E0308]: mismatched types
  --> tests/ui/compile_fail/wrong_type.rs:19:24
   |
19 |     let test: String = copyright_year7!(template, Cow::from("2023".to_string()));
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `String`, found `()`
